src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseConcreteTypeAnalyzer.State.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Concurrent;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.NetCore.Analyzers.Performance
+{
+    public partial class UseConcreteTypeAnalyzer
+    {
+        private sealed class State
+        {
+            public readonly ConcurrentDictionary<IFieldSymbol, bool> VirtualDispatchFields = new(SymbolEqualityComparer.Default);
+            public readonly ConcurrentDictionary<ILocalSymbol, bool> VirtualDispatchLocals = new(SymbolEqualityComparer.Default);
+            public readonly ConcurrentDictionary<IParameterSymbol, bool> VirtualDispatchParameters = new(SymbolEqualityComparer.Default);
+            public readonly ConcurrentDictionary<IMethodSymbol, bool> MethodsAssignedToDelegate = new(SymbolEqualityComparer.Default);
+
+            public readonly ConcurrentDictionary<IFieldSymbol, ConcurrentBag<ITypeSymbol>> FieldAssignments = new(SymbolEqualityComparer.Default);
```

Comment:
`ConcurrentBag` - can these also be `PooledHashSet` that are returned back to pool in the Dispose method?